viewing_party/main.py

```diff
@@ -0,0 +1,271 @@
+
+    Args:
+        movie_title (str): The title of a movie. Defaults to str.
+        genre (str): The movie genre. Defaults to str.
+        rating (float): The movie rating. Defaults to float.
+
+    Returns:
+        new_movie: A dictionary containing the keys and values
+            for a movie. Returns None for invalid arguments.
+    """
+
+    new_movie = {
+        "title": movie_title,
+        "genre": genre,
+        "rating": rating
+    }
+
+    if not movie_title or not genre or not rating:
+        return None
+    else:
+        return new_movie
+
+def add_to_watched(user_data, movie):
+    """Adds a ``movie`` to the user's ``watched`` movies list.
+
+    Args:
+        user_data (dict): Represents the user's ``watched`` list.
+        movie (dict): Represents a movie with a ``title``, ``genre``,
+            and ``rating``.
+
+    Returns:
+        user_data (dict): The user's data, with their ``watched``
+            movies list updated.
+    """
+
+    if movie not in user_data["watched"]:
+        user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    """Adds the ``movie`` to the user's ``watchlist``.
+
+    Args:
+        user_data (dict): Represents the user's ``watchlist``.
+        movie (dict): Represents a movie with a ``title``, ``genre``,
+            and ``rating``.
+
+    Returns:
+        user_data (dict): The user's data, with their ``watchlist``
+            updated.
+    """
+
+    if movie not in user_data["watchlist"]:
+        user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    """Allows the user to update their movies lists based
+    on a ``title`` they have watched.
+
+    Args:
+        user_data (dict): Represents the user's data, with their
+            lists of movies.
+        title (str): Represents the ``title`` of the movie
+            the user has watched.
+
+    Returns:
+        user_data (dict): The user's data, with their lists 
+            of movies updated.
+    """
+
+    for entry in user_data["watchlist"]:
+        if entry["title"] == title:
+            user_data["watchlist"].remove(entry)
+            add_to_watched(user_data, entry)
+    return user_data
+
+# test_wave_02
+
+def get_watched_avg_rating(user_data):
+    """Calculates the average rating of all movies in the user's ``watched`` movies list.
+
+    Args:
+        user_data (dict): Represents the user's data, with their
+            list of ``watched`` movies, rated.
+
+    Returns:
+        float: The average rating of all movies in the user's
+            ``watched`` movies list. Unrated movies default to 0.0.
+    """
+
+    user_movies = user_data["watched"]
+    movies_list_length = len(user_movies)
+    default_rating = 0.0
+    avg_rating = 0.0
+
+    if movies_list_length > 0:
+        for movie in user_movies:
+            avg_rating += movie["rating"]
+    else:
+        return default_rating
```

Comment:
You might consider dealing with returning the default rating first to deal with the edge case first, such as:
```suggestion
    if movies_list_length == 0:
        return None
        
   for movie in user_movies:
        avg_rating += movie["rating"]
```